satpy/readers/tropomi_l2.py

```diff
@@ -206,10 +206,31 @@ def get_dataset(self, ds_id, ds_info):
         file_key = ds_info.get('file_key', ds_id.name)
         data = self[file_key]
         data.attrs = self.get_metadata(data, ds_info)
-        fill = data.attrs.pop('_FillValue')
+        fill_value = data.attrs.get('_FillValue', np.float32(np.nan))
         data = data.squeeze()
-        data = data.where(data != fill)
+
+        # preserve integer data types if possible
+        if np.issubdtype(data.dtype, np.integer):
+            new_fill = fill_value
+        else:
+            new_fill = np.float32(np.nan)
+            data.attrs.pop('_FillValue', None)
+        good_mask = data != fill_value
+
+        scale_factor = data.attrs.get('scale_factor')
+        add_offset = data.attrs.get('add_offset')
+        if scale_factor is not None:
+            data = data * scale_factor + add_offset
+
+        data = data.where(good_mask, new_fill)
         data = self._rename_dims(data)
+
+        # drop coords whose units are not meters
+        drop_list = ['y', 'x', 'layer', 'vertices']
+        coords_exist = list(coord in data.coords for coord in drop_list)
+        if any(coords_exist):
+            data = data.drop(np.array(drop_list)[np.array(coords_exist)])
```

Comment:
This documentation says that `drop` is not the preferred method any more: http://xarray.pydata.org/en/stable/generated/xarray.DataArray.drop.html. But it also doesn't say that there is a drop specifically for `coords` either. Any idea what the best choice is? I was looking at the docs because I was wondering if there is anotehr way to do this line without converting to a numpy array first. Oh:

```
coords_exist = [coord for coord in drop_list if coord in data.coords]
if coords_exist:
    data = data.drop(coords_exist)
```